qa/integration/services/service_container.rb

```diff
@@ -0,0 +1,66 @@
+require_relative "service"
+require "docker"
+
+# Represents a service running within a container.
+class ServiceContainer < Service
+
+  def initialize(name, settings)
+    super(name, settings)
+
+    @base_image_context = File.expand_path("../dockerized", __FILE__)
+
+    @image_context = File.join(@base_image_context, @name)
+
+    # Options to create the container.
+    @container_create_opts = {}
+
+    # Options to delete the container.
+    @container_delete_opts = {:force => true, :volumes => true}
```

Comment:
@wainersm I think this shouldn't be a field and just be inlined to the `stop_container` method. Cleaning up volumes and `force` killing containers in the shutdown isn't optional.
Also, I don't think we should create a situation where a container has to "clean up" after itself ever. That will just get very confusing :)